service/user_http_test.go

```diff
@@ -82,3 +46,56 @@ func makeHTTPCall(method, path, body string, handlerFunc http.HandlerFunc) (reco
 	router.ServeHTTP(recorder, req)
 	return
 }
+
+func (suite *UsersHandlerTestSuite) TestRegisterUserSuccess() {
+	suite.dbMock.On("CreateUser", mock.Anything, mock.Anything).Return(db.User{}, nil)
+	suite.dbMock.On("GetUserByEmail", mock.Anything, mock.Anything).Return(db.User{}, sql.ErrNoRows)
+	body :=
+		`{
+			"first_name" : "test1",
+			"last_name" : "test2",
+			"email" : "test@gmail.com",
+			"mobile": "8421987856",
+			"country": "India",
+			"state": "Maharashtra",
+			"city": "Nashik",
+			"address": "abc",
+			"password": "password"
+		}`
+	recorder := makeHTTPCall(http.MethodPost,
+		"/register",
+		body,
+		registerUserHandler(Dependencies{Store: suite.dbMock}),
+	)
+	assert.Equal(suite.T(), http.StatusCreated, recorder.Code)
+	suite.dbMock.AssertExpectations(suite.T())
+
+}
+
+func (suite *UsersHandlerTestSuite) TestRegisterUserFailure() {
+	suite.dbMock.On("CreateUser", mock.Anything, mock.Anything).Return(db.User{}, nil)
```

Comment:
avoid using `mock.Anything`. 

you can you `mock.Anything` for `context.Context`